### PR TITLE
feat(config): a 3rd option to run the tests by dynamically loading test scripts without using iframe or new window

### PR DESCRIPTION
**Pull request by chan1cyrus2**
Implement a new option for user to run their test. The new option doesn't use iframe or a new window to run the tests. It combines both client.html and context.html files and then load the scripts dynamically when execute is received. The call flow remains the same as the other two options. This solution is used on our project that we need to support a lightweight browser which does not support both iframe and opening a new window. Since the scripts are loaded dynamically, the loading time is longer than the other two options.

### CODE REVIEW

**Comment by chan1cyrus2** [comment]
I signed it!

**Comment by zzo** [comment]
@dignifiedquire what do you think?

File: console.log [hunk 1]
```diff
@@ -1,5 +1,5 @@
-PhantomJS 2.1.1 (Mac OS X 0.0.0)
-PhantomJS 2.1.1 (Mac OS X 0.0.0)
-PhantomJS 2.1.1 (Mac OS X 0.0.0)
-PhantomJS 2.1.1 (Mac OS X 0.0.0)
-PhantomJS 2.1.1 (Mac OS X 0.0.0)
+PhantomJS 2.1.1 (Linux 0.0.0)
+PhantomJS 2.1.1 (Linux 0.0.0)
+PhantomJS 2.1.1 (Linux 0.0.0)
+PhantomJS 2.1.1 (Linux 0.0.0)
+PhantomJS 2.1.1 (Linux 0.0.0)
```
Comment (dignifiedquire) [inline console.log]: this seems like an unrelated change

Comment (dignifiedquire) [inline console.log]: Not even sure why this file exists in the first place tbh

File: static/client_with_context.html.html [hunk 2]
```diff
@@ -0,0 +1,125 @@
+
+<!DOCTYPE html>
```
Comment (dignifiedquire) [inline static/client_with_context.html.html]: `.html.html`?

Comment (chan1cyrus2) [inline static/client_with_context.html.html]: my bad

**Comment by chan1cyrus2** [comment]
Thanks!

I have added a e2e test case to cover the scenario (first time writing test case on cucumberjs, let me know if I do sth wrong) and updated the documentation on configuration.

I am not sure the part on overriding client_with_context.html in the same way the other templates can be overridden, can you elaborate more?

**Comment by chan1cyrus2** [comment]
I have added a e2e test case to cover the scenario (first time writing test case on cucumberjs, let me know if I do sth wrong) and updated the documentation on configuration.

I am not sure the part on overriding client_with_context.html in the same way the other templates can be overridden, can you elaborate more?

**Comment by dignifiedquire** [comment]
> I am not sure the part on overriding client_with_context.html in the same way the other templates can be overridden, can you elaborate more?

there is an config option `customContextFile`, which is used here: https://github.com/karma-runner/karma/blob/master/lib/middleware/karma.js#L146 to allow the user to customise the html of the actual file being loaded in the browser. It would be good to have this option for the new template as well.

**Comment by chan1cyrus2** [comment]
Thanks! I have added the option. Please review.
